.github/workflows/kb_sdk-actions.yml

```diff
@@ -0,0 +1,136 @@
+name: Basic kb-sdk tests
+
+on:
+  [push, pull_request]
+
+jobs:
+
+  test_kb-sdk_builds:
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        language: [perl, python, java]
+        test_type: [base, example]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Build with Ant
+        run: |
+          make
+
+      - name: Add bin to $PATH
+        run: |
+          echo "::add-path::$GITHUB_WORKSPACE/bin"
+
+      - name: checking kb-sdk functions
+        env:
+          KBASE_TEST_TOKEN: ${{ secrets.KBASE_TEST_TOKEN }}
+          LANGUAGE_TOKEN: ${{ matrix.language }}
+        run: |
+          env
+          kb-sdk help
+          kb-sdk version
+
+      - name: init base repo
+        if: matrix.test_type == 'base'
+        env:
+          LANGUAGE_TOKEN: ${{ matrix.language }}
+        run: |
+          kb-sdk init -l $LANGUAGE_TOKEN -u user SampleApp
+
+      - name: init example repo
+        if: matrix.test_type == 'example'
+        env:
+          LANGUAGE_TOKEN: ${{ matrix.language }}
+        run: |
+          kb-sdk init -l $LANGUAGE_TOKEN -u user --example SampleApp
+
+      - name: test ${{ matrix.test_type }} repo
+        env:
+          KBASE_TEST_TOKEN: ${{ secrets.KBASE_TEST_TOKEN }}
+          LANGUAGE_TOKEN: ${{ matrix.language }}
+        run: |
+          cd SampleApp
+          kb-sdk test || true
+          sed -i "s/test_token=/test_token=$KBASE_TEST_TOKEN/" test_local/test.cfg
+          kb-sdk test
+
+      - name: make resulting app available as artefact in case of failure
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: kbaseapp-${{ matrix.language }}-${{ matrix.test_type }}-${{ matrix.os }}
+          path: SampleApp
+
+
+  test_existing_repos:
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        app: [WsLargeDataIO, KBaseReport]
```

Comment:
Hmm... this means if those repos break then kb-sdk test also breaks. I understand why you're doing this but making the sdk build depend on outside repos building and testing correctly seems overly fragile.

Reply:
Also, neither of these repos test callback server functionality

Reply:
What would you suggest here? The tests are kind of moot anyway because the app base images may have a different version of kb-sdk on them, so the build/compile step won't really be testing the revised kb-sdk. There should be something to test that the changes aren't going to break existing repos.

Reply:
Well personally, I wouldn't test against outside repos and I would depend on a good suite of regression tests and quality PR reviews to ensure that changes don't break repos. So let's just make that happen yeah

Reply:
Hah! I have some ideas for setting up some better regression tests but I didn't want to put too much into this PR, especially since I am working around the knowledge that kb-sdk v1.1.0 to 1.2.0 introduces breaking changes in the python code which have been masked because the sdkbase repo has not been updated. 🤦

I really wish that the old testing mechanisms had not been disabled!

Reply:
Were they actually disabled? I just thought that no one could get them to run except me. I'm not even sure if I can get them to run at this point, so I wouldn't be surprised if bitrot had set in.

Reply:
I tried to get them going a while back, but they ran on a VM which has now disappeared, and the config/set up files were removed from the repo.

Reply:
> the config/set up files were removed from the repo

That's just... wow.